Test failures with yt-dlp 2026.03.17 and Python 3.15.0b2

### Checklist

- [x] I'm requesting a feature unrelated to a specific site
- [x] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [x] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [x] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=is%3Aissue%20-label%3Aspam%20%20) for similar requests **including closed ones**. DO NOT post duplicates
- [x] I've read the [policy against AI/LLM contributions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#automated-contributions-ai--llm-policy) and understand I may be blocked from the repository if it is violated

### Provide a description that is worded well enough to be understood

We are working on rebuilding all Python packages in Fedora Linux for Python 3.15.0b2. This part of our effort to catch issues and work with upstream projects to resolve them before new Python versions become GA.

There are currently a couple test failures related to PO tokens when running pytest with Python 3.15.0b2 and yt-dlp 2026.03.17. They are reproduced below. See https://kojipkgs.fedoraproject.org//work/tasks/2982/146232982/build.log for full logs (search for "/usr/bin/pytest" to find the pytest run).

### Provide verbose output that clearly demonstrates the problem

- [ ] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [ ] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
=================================== FAILURES ===================================
______________ TestPoTokenRequestDirector.test_clean_pot_generate ______________
self = <test_pot_director.TestPoTokenRequestDirector object at 0xffff8ee055b0>
ie = <yt_dlp.extractor.youtube.YoutubeIE object at 0xffff8cc6e030>
pot_request = PoTokenRequest(context=<PoTokenContext.GVS: 'gvs'>, innertube_context={'client': {'clientName': 'WEB'}}, innertube_hos...ne, request_headers={}, request_timeout=None, request_source_address=None, request_verify_tls=True, bypass_cache=False)
pot_cache = <test_pot_director.pot_cache.<locals>.MockPoTokenCache object at 0xffff927901a0>
logger = <conftest.MockLogger object at 0xffff8cc378e0>
    def test_clean_pot_generate(self, ie, pot_request, pot_cache, logger):
        # Token should be cleaned before returning
        base_token = base64.urlsafe_b64encode(b'token').decode()
        director = PoTokenRequestDirector(logger=logger, cache=pot_cache)
        provider = success_ptp(PoTokenResponse(base_token + '?extra=params'))(ie, logger, settings={})
        director.register_provider(provider)
    
        response = director.get_po_token(pot_request)
>       assert response == base_token
E       AssertionError: assert 'dG9rZW4extraparamg==' == 'dG9rZW4='
E         
E         - dG9rZW4=
E         + dG9rZW4extraparamg==
test/test_pot/test_pot_director.py:1096: AssertionError
_______________ TestPoTokenRequestDirector.test_clean_pot_cache ________________
self = <test_pot_director.TestPoTokenRequestDirector object at 0xffff8ed3dd90>
ie = <yt_dlp.extractor.youtube.YoutubeIE object at 0xffff8cc6d8b0>
pot_request = PoTokenRequest(context=<PoTokenContext.GVS: 'gvs'>, innertube_context={'client': {'clientName': 'WEB'}}, innertube_hos...ne, request_headers={}, request_timeout=None, request_source_address=None, request_verify_tls=True, bypass_cache=False)
pot_cache = <test_pot_director.pot_cache.<locals>.MockPoTokenCache object at 0xffff8eaceba0>
logger = <conftest.MockLogger object at 0xffff8d51be30>
pot_provider = <test_pot_director.success_ptp.<locals>.SuccessPTP object at 0xffff8eace900>
    def test_clean_pot_cache(self, ie, pot_request, pot_cache, logger, pot_provider):
        # Token retrieved from cache should be cleaned before returning
        base_token = base64.urlsafe_b64encode(b'token').decode()
        pot_cache.store(pot_request, PoTokenResponse(base_token + '?extra=params'))
        director = PoTokenRequestDirector(logger=logger, cache=pot_cache)
        director.register_provider(pot_provider)
    
        response = director.get_po_token(pot_request)
>       assert response == base_token
E       AssertionError: assert 'dG9rZW4extraparamg==' == 'dG9rZW4='
E         
E         - dG9rZW4=
E         + dG9rZW4extraparamg==
test/test_pot/test_pot_director.py:1111: AssertionError
___________________________ test_clean_pot_fail[123] ___________________________
po_token = '123'
    @pytest.mark.parametrize('po_token', [
        'invalid-token?',
        '123',
    ])
    def test_clean_pot_fail(po_token):
>       with pytest.raises(ValueError, match='Invalid PO Token'):
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E       Failed: DID NOT RAISE <class 'ValueError'>
test/test_pot/test_pot_director.py:1450: Failed
_ test_clean_pot[TwAA%5F%2D9VA6Q92v%5FvEQ4==?extra-param=2-TwAA_-9VA6Q92v_vEQ4=] _
po_token = 'TwAA%5F%2D9VA6Q92v%5FvEQ4==?extra-param=2'
expected = 'TwAA_-9VA6Q92v_vEQ4='
    @pytest.mark.parametrize('po_token,expected', [
        ('TwAA/+8=', 'TwAA_-8='),
        ('TwAA%5F%2D9VA6Q92v%5FvEQ4==?extra-param=2', 'TwAA_-9VA6Q92v_vEQ4='),
    ])
    def test_clean_pot(po_token, expected):
>       assert clean_pot(po_token) == expected
E       AssertionError: assert 'TwAA_-9VA6Q9...extra-param0=' == 'TwAA_-9VA6Q92v_vEQ4='
E         
E         - TwAA_-9VA6Q92v_vEQ4=
E         + TwAA_-9VA6Q92v_vEQ4extra-param0=
E         ?                    ++++++++++++
test/test_pot/test_pot_director.py:1459: AssertionError
=============================== warnings summary ===============================
test/test_pot/test_pot_director.py::test_clean_pot[TwAA/+8=-TwAA_-8=]
  /builddir/build/BUILD/yt-dlp-2026.03.17-build/yt-dlp-2026.03.17/yt_dlp/extractor/youtube/pot/_director.py:438: FutureWarning: invalid character '+' in URL-safe Base64 data will be discarded in future Python versions
    base64.urlsafe_b64decode(urllib.parse.unquote(po_token))).decode()
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Running in a local venv with 3.15.0b2 and acf8ab7a6e3024325f62426e35a17f365c4d5d54, I get

```
================================================== FAILURES ==================================================
__________________________________________ test_clean_pot_fail[123] __________________________________________
test/test_pot/test_pot_director.py:1450: in test_clean_pot_fail
    with pytest.raises(ValueError, match='Invalid PO Token'):
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E   Failed: DID NOT RAISE <class 'ValueError'>
========================================== short test summary info ===========================================
FAILED test/test_pot/test_pot_director.py::test_clean_pot_fail[123] - Failed: DID NOT RAISE <class 'ValueError'>
```

--- a/yt_dlp/extractor/youtube/pot/_director.py
+++ b/yt_dlp/extractor/youtube/pot/_director.py
@@ -438,9 +438,11 @@ def clean_pot(po_token: str):
     if not mobj:
         raise ValueError('Invalid PO Token')
 
+    # compat: <=py3.14: padded was added in 3.15 and the default for urlsafe is false
+    data = mobj.group(1).translate(str.maketrans('-_', '+/'))
     try:
         return base64.urlsafe_b64encode(
-            base64.urlsafe_b64decode(mobj.group(1))).decode()
+            base64.b64decode(data)).decode()
     except (binascii.Error, ValueError):
         raise ValueError('Invalid PO Token')
 

